src/txn/txn_recover.c

```diff
@@ -781,6 +786,7 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
          * earlier time.
          */
         WT_ERR(__recovery_file_scan(&r));
+        recovery_file_scanned = true;
```

Comment:
Instead of `recovery_file_scanned` the condition here says that there is no log recovery. How about a variable name of `log_recovery` and set it to true by default and mark it false in this code flow?